### PR TITLE
Support parent death signal

### PR DESCRIPTION
**Pull request by StevenJiang1110**

### CODE REVIEW

**Comment by tatetian** [comment]
@StevenJiang1110 Need to rebase this PR

**Comment by tatetian** [comment]
@cchanging Could you review this PR? Thanks a lot!

File: kernel/aster-nix/src/process/posix_thread/exit.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+fn is_main_thread(tid: Tid, posix_thread: &PosixThread) -> bool {
+    let process = posix_thread.process();
+    let pid = process.pid();
+    tid == pid
+}
+
+fn is_last_thread(thread: &PosixThread) -> bool {
+    let process = thread.process.upgrade().unwrap();
```
Comment (cchanging) [inline kernel/aster-nix/src/process/posix_thread/exit.rs:59]: Can these methods like `is_main_thread` and `is_last_thread` be methods of `PosixThread`? These function names sound more like they should be methods.

File: kernel/aster-nix/src/syscall/prctl.rs [hunk 2]
```diff
@@ -6,15 +6,35 @@
...
     match prctl_cmd {
+        PrctlCmd::PR_SET_PDEATHSIG(signum) => {
+            let current = current!();
+            current.set_parent_death_signal(signum);
+        }
+        PrctlCmd::PR_GET_PDEATHSIG(write_to_addr) => {
+            let write_val = {
+                let current = current!();
+
+                match current.parent_death_signal() {
```
Comment (cchanging) [inline kernel/aster-nix/src/syscall/prctl.rs:30]: Does the use of the API here cause a race condition? For example, if another thread of the same process resets `parent_death_signal` value between obtaining the current `parent_death_signal` and `write_val_to_user`. If this situation does not occur or is acceptable, I think `parent_death_signal` might be implemented using `AtomicU8`, with the value 0 corresponding to `None`.

**Review by cchanging — COMMENTED** [review]
Good implementation! Here are some of my suggestions.

**Review by cchanging — APPROVED** [review]
LGTM!

**Pull request merged**